crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -250,4 +252,93 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    /// Asserts that a subscription holds a tx handle for the entire length of its evaluation.
+    fn test_subscriptions_for_the_same_client_identity() -> ResultTest<()> {
+        let test_db = TestDB::durable()?;
+        let runtime = test_db.runtime().cloned().unwrap();
+
+        // Create table with no rows
+        let db = Arc::new(test_db.db.clone());
+        let table_id = db.create_table_for_test("T", &[("a", AlgebraicType::U8)], &[])?;
+
+        let id = ClientActorId::for_test();
+        let sender = Arc::new(ClientConnectionSender::dummy(id, Protocol::Binary));
+        let module_subscriptions = ModuleSubscriptions::new(db.clone(), id.identity);
+
+        let client_id0 = ClientActorId::for_test();
+        let (client0, mut rx0) = ClientConnectionSender::dummy_with_channel(client_id0, Protocol::Binary);
+
+        let client_id1 = ClientActorId::for_test();
+        let (client1, mut rx1) = ClientConnectionSender::dummy_with_channel(client_id1, Protocol::Binary);
```

Comment:
Does this test what you want? Won't the Identities be different here?

Reply:
you're right, no idea what I was thinking 😅 I'll push a fix